src/hooks/usePairTokenPriceByTimestamp.ts

```diff
@@ -0,0 +1,52 @@
+import { Currency, Pair, Token } from '@swapr/sdk'
+
+import { useEffect, useState } from 'react'
+
+import {
+  convertToChartData,
+  GetBlockPairTokenPriceQueryData,
+  TIMEFRAME_PROPRETIES,
+} from '../components/Charts/chartUtils'
+import { useGetPairTokenPricesQuery } from '../graphql/generated/schema'
+import { wrappedCurrency } from '../utils/wrappedCurrency'
+
+import { useActiveWeb3React } from '.'
+
+type PairTokenPriceByTimestampProps = {
+  currency0?: Currency
+  currency1?: Currency
+  dateInterval: string
+}
+
+export function usePairTokenPriceByTimestamp({ currency0, currency1, dateInterval }: PairTokenPriceByTimestampProps) {
+  const { chainId } = useActiveWeb3React()
+  const [wrappedToken1, setWrappedToken1] = useState<Token>()
+  const [pairAddress, setPairAddress] = useState<string>()
+
+  const { data, loading, error } = useGetPairTokenPricesQuery({
+    variables: {
+      pairAddress,
+      timestamp: TIMEFRAME_PROPRETIES[dateInterval].timestamp,
+      timeframe: TIMEFRAME_PROPRETIES[dateInterval].pairTokenPriceTimeframe,
+    },
+  })
+
+  useEffect(() => {
+    try {
+      const wrappedToken0 = wrappedCurrency(currency0, chainId)
+      const wrappedToken1 = wrappedCurrency(currency1, chainId)
```

Comment:
what do you think abouti it ?
```suggestion
      const [wrappedToken0, wrappedToken1] = [wrappedCurrency(currency0, chainId), wrappedCurrency(currency1, chainId)]
```

Reply:
Interesting. I'll give it a shot. As they come in pairs this might be an interesting approach.